tmd/Topology/methods.py

```diff
@@ -39,10 +39,10 @@ def get_persistence_diagram(tree, feature='radial_distances', **kwargs):
     children = {b: end[np.where(beg == b)[0]] for b in np.unique(beg)}
 
     while len(np.where(active)[0]) > 1:
-        alive = list(np.where(active)[0])
-        for l in alive:
+        alives = list(np.where(active)[0])
```

Comment:
how big is `active`, b/c `np.where` needs to read the whole thing twice, instead of just calculating it once

Reply:
I think this is some very old code, which for some reason was not complying with pylint anymore, I just made it pass lint, but  I agree, it can be improved!

Reply:
Active should be a very small list in principle. Is there an alternative to np.where that is more optimal though? (It's indeed very old code here).

Reply:
I think you could just do this:
```python
    alives = np.where(active)[0]
    while len(alives) > 1:
        for alive in alives:
            .......
        alives = np.where(active)[0]
```
Tests pass with this.